In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as st

In [ ]:
sales=pd.read_csv('/content/drive/MyDrive/Data Science/supermarket_sales - Sheet1.csv')
sales

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating  
0         26.1415     9.1  
1          3.8200     9.6  
2         16.2155     7.4  
3         23.2880     8.4  
4         30.2085     5.3  
..            ...     ...  
995        2.0175     6.2  
996       48.6900     4.4  
997        1.5920     7.7  
998        3.2910     4.1  
999       30.9190     6.6  

[1000 rows x 17 columns]

In [ ]:
sales.columns

Index(['Invoice ID', 'Branch', 'City', 'Customer type', 'Gender',
       'Product line', 'Unit price', 'Quantity', 'Tax 5%', 'Total', 'Date',
       'Time', 'Payment', 'cogs', 'gross margin percentage', 'gross income',
       'Rating'],
      dtype='object')

In [ ]:
df=pd.DataFrame({'Branch':sales['Branch'],'Gender':sales['Gender'],'Product':sales['Product line'],'Total':sales['Total'],'Paymnet':sales['Payment'],})
df['Customer type']=sales['Customer type']
df

Branch  Gender                 Product      Total      Paymnet  \
0        A  Female       Health and beauty   548.9715      Ewallet   
1        C  Female  Electronic accessories    80.2200         Cash   
2        A    Male      Home and lifestyle   340.5255  Credit card   
3        A    Male       Health and beauty   489.0480      Ewallet   
4        A    Male       Sports and travel   634.3785      Ewallet   
..     ...     ...                     ...        ...          ...   
995      C    Male       Health and beauty    42.3675      Ewallet   
996      B  Female      Home and lifestyle  1022.4900      Ewallet   
997      A    Male      Food and beverages    33.4320         Cash   
998      A    Male      Home and lifestyle    69.1110         Cash   
999      A  Female     Fashion accessories   649.2990         Cash   

    Customer type  
0          Member  
1          Normal  
2          Normal  
3          Member  
4          Normal  
..            ...  
995        Normal  
996        Normal  
997        Member  
998        Normal  
999        Member  

[1000 rows x 6 columns]

In [ ]:
df.groupby('Branch').mean()

Total
Branch            
A       312.354031
B       319.872506
C       337.099715

# Hypothesis testing

## H0: Branch C makes an averge total of 337 Dollars per day.
## H1: Branch C makes more than 337 Dollars per day.

In [ ]:
sample=df.loc[df['Branch']=='C']
samplec=sample.sample(165)
samplec

Branch  Gender                 Product     Total      Paymnet  \
600      C    Male  Electronic accessories   87.2340      Ewallet   
897      C  Female      Food and beverages  297.8010         Cash   
711      C  Female      Home and lifestyle  317.2260      Ewallet   
237      C  Female     Fashion accessories  232.1550  Credit card   
383      C  Female      Food and beverages  314.5380      Ewallet   
..     ...     ...                     ...       ...          ...   
295      C  Female  Electronic accessories  272.5800         Cash   
828      C    Male  Electronic accessories  681.9750         Cash   
102      C  Female  Electronic accessories  218.0115         Cash   
420      C  Female      Food and beverages  121.5900      Ewallet   
236      C    Male       Health and beauty   91.4025  Credit card   

    Customer type  
600        Normal  
897        Member  
711        Member  
237        Member  
383        Member  
..            ...  
295        Member  
828        Normal  
102        Normal  
420        Member  
236        Normal  

[165 rows x 6 columns]

In [ ]:
samplec.Total.mean()

## z test

In [ ]:
#standard deviation of the population
sd=np.std(samplec['Total'])
sd

282.14941702026755

In [ ]:
samplec.std()

<ipython-input-115-4d62aafc4dea>:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  samplec.std()


Total    283.008321
dtype: float64

In [ ]:
xbar=samplec.Total.mean()
mu=337.09
n=165 #sample size
alpha=0.05

In [ ]:
xbar

336.8119363636364

In [ ]:
zvalue=(xbar-mu)/(sd/np.sqrt(n))
zvalue

-0.01361299693475027

In [ ]:
z_critical = st.norm.ppf(alpha)  #One tail Test
z_critical

-1.6448536269514729

In [ ]:
if (zvalue<z_critical and abs(zvalue)> abs(z_critical)):
  print("Reject Null Hypotheisis")
else:
  print("Fail to reject Null Hypothesis")

Fail to reject Null Hypothesis


## using p value

In [ ]:
p_val = (1 - st.norm.cdf(abs(zvalue)))
p_val

0.1724861238457459

In [ ]:
#comparing with alpha value
if p_val<alpha:
  print("Reject Null Hypotheisis")
else:
  print("Fail to reject Null Hypothesis")

Fail to reject Null Hypothesis


## Hence Branch C On Average doesn't make more than 337$ per day, i.e, we fail to reject the Null Hypothesis

## T test

In [ ]:
tsamplec=sample.sample(25)
tsamplec

Branch  Gender                 Product     Total      Paymnet  \
617      C    Male  Electronic accessories  461.5275      Ewallet   
259      C    Male  Electronic accessories  276.9480  Credit card   
343      C  Female      Food and beverages  509.4075      Ewallet   
61       C    Male      Home and lifestyle  351.0990      Ewallet   
923      C  Female       Sports and travel  611.0790  Credit card   
726      C    Male      Home and lifestyle  175.0455  Credit card   
5        C    Male  Electronic accessories  627.6165      Ewallet   
619      C  Female     Fashion accessories  273.7980      Ewallet   
906      C    Male       Health and beauty  888.4050  Credit card   
401      C    Male      Home and lifestyle  194.1240  Credit card   
936      C  Female       Health and beauty  579.8415      Ewallet   
121      C    Male       Sports and travel  419.8320  Credit card   
109      C    Male  Electronic accessories  860.6850         Cash   
368      C  Female       Sports and travel  150.7800         Cash   
420      C  Female      Food and beverages  121.5900      Ewallet   
314      C  Female  Electronic accessories  820.3650         Cash   
252      C  Female       Sports and travel   30.2190  Credit card   
935      C    Male      Home and lifestyle  270.0180      Ewallet   
766      C  Female     Fashion accessories  239.5890  Credit card   
160      C  Female      Food and beverages  362.7120  Credit card   
317      C    Male  Electronic accessories  392.6475         Cash   
983      C    Male       Health and beauty  734.7060         Cash   
897      C  Female      Food and beverages  297.8010         Cash   
822      C    Male       Sports and travel   10.6785         Cash   
50       C    Male      Food and beverages  671.7900         Cash   

    Customer type  
617        Member  
259        Member  
343        Normal  
61         Normal  
923        Normal  
726        Member  
5          Normal  
619        Member  
906        Normal  
401        Normal  
936        Normal  
121        Member  
109        Member  
368        Normal  
420        Member  
314        Member  
252        Normal  
935        Member  
766        Normal  
160        Normal  
317        Member  
983        Normal  
897        Member  
822        Member  
50         Member

In [ ]:
t_sd=np.std(tsamplec['Total'])
t_sd

250.01686218164883

In [ ]:
xbar=tsamplec.Total.mean()
print(xbar)
mu=337.09
n=25 #sample size
alpha=0.05

413.2921800000001


In [ ]:
n = 25
degrees_of_freedom = n-1
t = (xbar - mu)/(t_sd / np.sqrt(n))
t

1.523940812132817

In [ ]:
t_critical = st.t.ppf(alpha, degrees_of_freedom)
t_critical

-1.7108820799094282

In [ ]:
if (t > t_critical):
    print("Null hypothesis cannot be rejected")
else:
    print("Reject null hypothesis")

Null hypothesis cannot be rejected


## Both Z test and T test suggest the same result, The Null Hypothesis cannot be rejected.